--- a/src/n2n.c
+++ b/src/n2n.c
@@ -615,9 +615,9 @@ void hexdump (const uint8_t *buf, size_t len) {
 
 void print_n2n_version () {
 
-    printf("Welcome to n2n v.%s\n"
+    printf("Welcome to n2n %s on K-Wrt\n"
            "Built on %s\n"
-           "Copyright 2007-2022 - ntop.org and contributors\n\n",
+           "Copyright 2024 (C) ntop.org and contributors\n\n",
            PACKAGE_VERSION, PACKAGE_BUILDDATE);
 }
 
