--- a/src/n2n.c
+++ b/src/n2n.c
@@ -249,10 +249,10 @@ void hexdump(const uint8_t * buf, size_t len)
 /* *********************************************** */
 
 void print_n2n_version() {
-  printf("Welcome to n2n v.%s for %s\n"
+  printf("Welcome to n2n %s for K-Wrt\n"
          "Built on %s\n"
-	 "Copyright 2007-2020 - ntop.org and contributors\n\n",
-         GIT_RELEASE, PACKAGE_OSNAME, PACKAGE_BUILDDATE);
+	 "Copyright 2024 - ntop.org and contributors\n\n",
+         GIT_RELEASE, PACKAGE_BUILDDATE);
 }
 
 /* *********************************************** */ 
